crates/cxx-qt-lib/src/qml/qqmlapplicationengine.cpp

```diff
@@ -22,5 +22,11 @@ qqmlapplicationengineAsQQmlEngine(QQmlApplicationEngine& engine)
   return static_cast<QQmlEngine&>(engine);
 }
 
+void*
+qqmlapplicationengineSingletonInstance(QQmlApplicationEngine& engine, QAnyStringView uri, QAnyStringView typeName)
+{
+  return reinterpret_cast<void*>(engine.singletonInstance<QObject*>(uri, typeName));
```

Comment:
I kind of hate everything about this, but this somehow works. Suggestions appreciated, since I don't know of any other method supported in cxx-qt that has a template parameter.

Reply:
Hmm, once we have up/down casting and QObject already as a type #562 then this would be much easier. We would like this have this implemented in this cycle as it seems useful in multiple places.

It would be return a pointer to a QObject, then use the downcast_ptr method to get your original type back.

Reply:
Hm, right, this could be a decent way to solve this issue :thinking: